packages/batch-submitter/src/exec/run-batch-submitter.ts

```diff
@@ -15,19 +15,45 @@ import {
   TX_BATCH_SUBMITTER_LOG_TAG,
 } from '..'
 
+const environment = process.env.NODE_ENV
+const release = `batch-submitter@${process.env.npm_package_version}`
+
 /* Logger */
 const name = 'oe:batch_submitter:init'
-const log = new Logger({
-  name,
-  sentryOptions: {
-    release: `batch-submitter@${process.env.npm_package_version}`,
-    dsn: process.env.SENTRY_DSN,
-    tracesSampleRate: 0.05,
-  },
-})
+let log
+switch (environment) {
+  case 'kovan':
+  case 'mainnet':
+    log = new Logger({
+      name,
+      sentryOptions: {
+        release,
+        dsn: process.env.SENTRY_DSN,
+        tracesSampleRate: 0.05,
+        environment,
+      },
+    })
+    break
+  case 'goerli':
+    log = new Logger({
+      name,
+      sentryOptions: {
+        release,
+        dsn: process.env.SENTRY_DSN,
+        tracesSampleRate: 1, // Use 100% trace rate
+        environment,
+      },
+    })
+    break
+  default:
+    // Skip initializing Sentry
+    log = new Logger({ name })
+}
+
 /* Metrics */
 const metrics = new Metrics({
   prefix: name,
+  labels: { environment, release },
 })
```

Comment:
@snario if this logic looks reasonable, i plan to do this for the dtl by moving logger and metrics initialization from the constructor to `init()` below this code block:

https://github.com/ethereum-optimism/optimism/blob/4c2fbdeac31908e240774723ba69a6a381481d47/packages/core-utils/src/base-service.ts#L24-L39

and allow for an optional `monitorOpts` parameter in `init` so we can pass in the names + environments and more. that would be a bigger refactor so i'd get it in a followup PR to this one; lmk if you have other ideas though!

Reply:
Hm, personally I'd prefer if we moved _away_ from this generic "base service" template, but if we stick with it this seems reasonable.

Reply:
@snario any way we can do that piece by piece? to me it seems like it's either we do migrate off all at once or not based on the way this is structured, so i've been defaulting to not doing anything since i'm not super opinionated here